libkbfs/folder_branch_ops.go

```diff
@@ -1206,6 +1206,15 @@ func (fbo *folderBranchOps) initMDLocked(
 		return err
 	}
 
+	fbo.headLock.Lock(lState)
+	defer fbo.headLock.Unlock(lState)
+	// Some other thread got here first, so give up and let it go
+	// before we push anything to the servers.
```

Comment:
Ah I see! Because the check against `ImmutableRootMetadata` was after calling the `Put`. Is the `headLock` here still necessary then? (although I guess it wouldn't hurt since internally it has a fast path using atomic compare and swap)